[WIP] Gutenboarding: restyle SignupForm to match current designs

andrewserong opened the pull request:
#### Changes proposed in this Pull Request

* Restyle signup form in Gutenboarding to match current Figma
* Add additional mixins to cover most of the common text sizes

Note: it is yet to be determined whether or not a password field is required, so I've left it off for the time being and just focused on styling in this PR.

Also note, that due to how the Modal component works, I've had to use `display:none` on the standard header that's included in the modal, and add our own custom header component. This is slightly hacky because there are technically two header components being mounted. My hope is that `display:none` is a suitable workaround to avoid writing our own modal from scratch, since the `wordpress/components` has some nice affordances.

#### Testing instructions

![image](https://user-images.githubusercontent.com/14988353/77886224-352c5900-72b4-11ea-958d-ba86c3ec80ed.png)

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* From a logged out session, go to `http://calypso.localhost:3000/gutenboarding and complete the signup process until you get to the `Save your progress` screen.
* Ensure closing the modal works as intended

Fixes #40504

andrewserong commented:
@Automattic/dotcom-create-design — can I please get your input on this. I've been working on restyling the signup form in Gutenboarding to match the Figma, however I'm not quite sure what to do about styling the `Notice` component that we use to display error messages (like for existing email addresses). Here's what that component looks like without any styling overrides:

![image](https://user-images.githubusercontent.com/14988353/77886668-01056800-72b5-11ea-9bb5-7152b9c04994.png)

Where do you think this error message would be best placed, and how should it be styled?

Thanks!

p-jackson commented on client/landing/gutenboarding/components/signup-form/header.tsx:
```diff
@@ -0,0 +1,61 @@
+
+const CustomCloseButton = ( { onClose, closeLabel }: CloseButtonProps ) => {
+	const { __: NO__ } = useI18n();
+	const label = closeLabel ? closeLabel : NO__( 'Close dialog' );
```
Was thinking about how we could use the new nullish coalescing operator for this, which is pretty cool (let's component users set the label to an empty string I guess)

```suggestion
	closeLabel = closeLabel ?? NO__( 'Close dialog' );
```

But actually I don't think we should have the `closeLabel` prop. `<CustomCloseButton>` isn't exported so I don't think it needs to be generally applicable.

andrewserong replied:
Good call. I was following the component used in the existing close button in the `Modal` component. I'll remove it.

p-jackson commented on client/landing/gutenboarding/components/signup-form/index.tsx:
```diff
@@ -101,46 +102,55 @@ const SignupForm = ( { onRequestClose }: Props ) => {
 			onRequestClose={ closeModal }
 			focusOnMount={ false }
-			isDismissible={ ! isFetchingNewUser }
+			isDismissible={ false }
```
Not sure about what `isDismissible={ false }` is doing because pressing Esc is definitely still working. We do what the modal dismissible right?

andrewserong replied:
Oh, this is a hack that just ensures that the `Modal` component's default close button isn't rendered. We then use our own. I think `isDismissible` is more accurately a `showCloseButton` feature.

p-jackson replied:
Ah! Thanks for that. I do think that you showing our own header, rather than bending over backwards with hacky CSS change the existing one do what we want, is a good idea.

p-jackson commented on client/landing/gutenboarding/components/signup-form/index.tsx:
```diff
@@ -101,46 +102,55 @@ const SignupForm = ( { onRequestClose }: Props ) => {
+				loginUrl={ '/log-in?redirect_to=' + encodeURIComponent( loginRedirectUrl ) }
+			/>
+
+			<main className="signup-form__body">
```
I don't think `<main>` is appropriate here. You're only allowed one on the page at a time, and it would normally be the non-header/footer, non-modal, main content of the page.

We're not getting an a11y violation warning in the browser, because Gutenboarding doesn't already have a `<main>` element, but it probably should.

You might have been looking for something like `<dialog>` or `<div role="dialog">`, but never fear, `<Modal>` has already taken care of that for you. So I think a `<div>` would be fine in this case.

andrewserong replied:
Oh, thanks — yes, `dialog` is more in the direction of what I was thinking. Thanks for looking into this, I'll switch to a `div`!

p-jackson commented on client/landing/gutenboarding/components/signup-form/style.scss:
```diff
@@ -1,26 +1,71 @@
+		}
 	}
 
 	.signup-form__error-notice {
```
My 2c on the error notice design: changing this rule to `margin: 0 30px` on desktop (and something else on mobile) is probably enough. Makes it line up with the text input, but that's it.

p-jackson replied:
GH hasn't made it super clear, but I'm referring to the `.signup-form__error-notice` rules.

andrewserong replied:
Great idea, that looks so much nicer.

andrewserong commented:
Thanks for reviewing @p-jackson — I've implemented your feedback, and updated the mobile styling to better match the Figma (so that it doesn't centre vertically — this way when the other steps are styled for mobile it should all line up correctly).

I think I might leave it here for this PR if it looks okay to you, and add in the password field in a separate PR so that this one doesn't grow too large.

andrewserong commented:
For completeness: I've merged this in and will continue with tweaks in other PRs (e.g. adding in the password field). The error message currently looks like this: (thanks for the margin suggestion, Phil, and the quick approval Ramon!)

![image](https://user-images.githubusercontent.com/14988353/77973442-f348f480-733f-11ea-9218-c7b1d4cfb168.png)

dubielzyk commented:
@andrewserong 
> I've been working on restyling the signup form in Gutenboarding to match the Figma, however I'm not quite sure what to do about styling the Notice component

I've added another mock in Figma, but here's a screenshot of how we should restyle it.
![image](https://user-images.githubusercontent.com/2896062/78010434-a74b7f00-7342-11ea-83de-d7fc9f2ce9a2.png)

- Plain centered text with `red60` and `font-weight: 400`
- Removed the _Create account_ button and terms as they don't make sense. Added _Already have an account? Log in_ link to redirect them
